service/ui.go

```diff
@@ -316,6 +316,8 @@ func (uis *UIServer) GetServiceApp() *gimlet.APIApp {
 	app.AddRoute("/json/task_log/{task_id}/{execution}").Wrap(needsLogin, needsContext, viewLogs).Handler(uis.taskLog).Get()
 	app.AddRoute("/task_log_raw/{task_id}/{execution}").Wrap(needsLogin, needsContext, allowsCORS, viewLogs).Handler(uis.taskLogRaw).Get()
 
+	// Proxy downloads for task uploaded files via S3
+	app.AddRoute(("/task_file_raw/{task_id}/{execution}/{file_name}")).Wrap(needsLogin, needsContext, allowsCORS, viewLogs).Handler(uis.taskFileRaw).Get()
```

Comment:
is it possible for filenames to be wonky such that a query parameter would be chaotic? (we could have this be part of the body instead if so)

Reply:
Nope as long as the URLs are properly encoded by whatever service is making the request this isn't an issue. The HTTP client handles decoding them automatically.